modules/autofill/autofill-impl/src/main/java/com/duckduckgo/autofill/AutofillJavascriptInterface.kt

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright (c) 2022 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.autofill
+
+import android.webkit.JavascriptInterface
+import android.webkit.WebView
+import com.duckduckgo.app.di.AppCoroutineScope
+import com.duckduckgo.app.email.EmailManager
+import com.duckduckgo.app.global.DefaultDispatcherProvider
+import com.duckduckgo.app.global.DispatcherProvider
+import com.duckduckgo.app.utils.ConflatedJob
+import com.duckduckgo.autofill.domain.app.LoginCredentials
+import com.duckduckgo.autofill.domain.javascript.JavascriptCredentials
+import com.duckduckgo.autofill.jsbridge.AutofillMessagePoster
+import com.duckduckgo.autofill.jsbridge.request.AutofillRequestParser
+import com.duckduckgo.autofill.jsbridge.request.SupportedAutofillInputMainType.CREDENTIALS
+import com.duckduckgo.autofill.jsbridge.response.AutofillResponseWriter
+import com.duckduckgo.autofill.store.AutofillStore
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import timber.log.Timber
+
+class AutofillJavascriptInterface(
```

Comment:
I think here we can use @ContributesBinding. I am not sure why we have  var callback: Callback? = null in the constructor instead of a local public variable.

Reply:
Fixed and fixed.

Reply:
I don't have full context but just wanted to leave a warning here in case is useful based on things I've seen before with other JS interfaces behaving weirdly. Just be careful with the instance not being a singleton because you definitely want one per tab :D.